inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java

```diff
@@ -73,21 +74,23 @@ public SortSourceConfigResponse getSourceConfig(String clusterName, String sortT
     }
 
     @Override
-    public ListSortStatusResponse listSortStatus(ListSortStatusRequest request) {
-        Preconditions.checkNotNull(sortStatusPoller, "sort job status poller not initialized");
+    public List<SortStatusInfo> listSortStatus(SortStatusRequest request) {
+        Preconditions.checkNotNull(sortPoller, "sort status poller not initialized, please try later");
         List<InlongGroupInfo> groupInfoList = request.getInlongGroupIds().stream()
-                .map(groupId -> groupService.get(groupId)).collect(Collectors.toList());
-        Map<String, SortStatus> statusMap = sortStatusPoller.poll(groupInfoList, request.getCredentials());
-        log.debug("get sort status map = {}", statusMap);
-        return ListSortStatusResponse.builder().statusMap(statusMap).build();
+                .map(groupId -> groupService.get(groupId))
+                .collect(Collectors.toList());
+
+        List<SortStatusInfo> statusInfos = sortPoller.pollSortStatus(groupInfoList, request.getCredentials());
+        log.debug("success list sort status for request={}, result={}", request, statusInfos);
+
+        return statusInfos;
     }
 
     @Override
     public void acceptPlugin(Plugin plugin) {
-        if (!(plugin instanceof ProcessPlugin)) {
-            return;
+        if (plugin instanceof PollerPlugin) {
+            PollerPlugin pollerPlugin = (PollerPlugin) plugin;
+            sortPoller = pollerPlugin.getSortPoller();
         }
-        ProcessPlugin processPlugin = (ProcessPlugin) plugin;
-        sortStatusPoller = processPlugin.createSortStatusPoller();
     }
```

Comment:
plugin only support a class from define of plugin.yaml. You can't add `FlinkSortPollerPlugin`. Because It can't be loaded to execute `acceptPlugin` method.

Reply:
Change the `plugin.yaml` to support loading more than one class in the manager plugin.